Docs/new external lib document (#172)
usebruno/bruno-docs: src/pages/testing/script/external-libraries.mdx

import { Callout } from 'nextra/components'
import { FileTree } from 'nextra/components'

# External Libraries

Bruno allows you to load any npm module for use in your scripting workflows. 

## Prerequisites
Before you begin, ensure that you have the following:

1. Node.js installed on your system.
2. Bruno installed.

## Project Structure

> The folder structure for this example will look as follows:

Reviewer: ```suggestion
The folder structure for this example will be:
```
```suggestion
The folder structure for this example will look as follows:
```

        <FileTree.File name="openspec.yaml" />
        <FileTree.File name=".spectral.yml" />
        <FileTree.File name="package.json" />
        <FileTree.Folder name="node_modules">
          <FileTree.File name="spectral-cli" />
        </FileTree.Folder>
      </FileTree.Folder>
    </FileTree>
 
## Configuration
You need to initialize `package.json` file inside your collection.

Navigate to your Bruno collection directory.
 ```bash
 cd path/to/your/bruno-collection
 ```

Initialize `package.json` file inside your collection.
```bash
npm init -y
```

Now you can install any npm package or module to use in your collection.

## Spectral Integration

In this example we're using [Spectral CLI](https://github.com/stoplightio/spectral?tab=readme-ov-file) as a npm package which help you to valide OpenSpec API YAML or JSON files with the built-in ruleset. 

You need to have OpenAPI JSON or YAML file to work with spectral or you can download from [here](https://github.com/ganesh-bruno/spectral-example) as example.

### Spectral CLI Setup
You can install `spectral-cli` by running following command:
```bash
npm install @stoplight/spectral-cli
```

> Run this command into your terminal to initialize `.spectral.yml` file which is predefined ruleset.

Reviewer: ```suggestion
Run this command in your terminal to initialize `.spectral.yml` file which is predefined ruleset.
```
